tests/unit/backend/chalice/api_server/test_v1_collection.py

```diff
@@ -427,8 +430,143 @@ def test__list_collection__check_owner(self):
+        test_url = furl(path=f"/dp/v1/collections/{collection.id}")
+        test_url.add(query_params=dict(visibility="PRIVATE"))
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url.url, headers=headers)
+        self.assertEqual(response.status_code, 202)
+
+    def test_delete_collection__public(self):
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PUBLIC.name, owner="test_user_id"
+        )
+        test_url = furl(path=f"/dp/v1/collections/{collection.id}")
+        test_url.add(query_params=dict(visibility="PRIVATE"))
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url.url, headers=headers)
+        self.assertEqual(response.status_code, 403)
+
+    def test_delete_collection__not_owner(self):
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="someone_else"
+        )
+        test_url = furl(path=f"/dp/v1/collections/{collection.id}")
+        test_url.add(query_params=dict(visibility="PRIVATE"))
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url.url, headers=headers)
+        self.assertEqual(response.status_code, 403)
+
+    def test_delete_collection__does_not_exist(self):
+        fake_uuid = str(uuid.uuid4())
+        test_url = furl(path=f"/dp/v1/collections/{fake_uuid}")
+        test_url.add(query_params=dict(visibility="PRIVATE"))
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url.url, headers=headers)
+        self.assertEqual(response.status_code, 403)
+
+    def test_deleted_collection_does_not_appear_in_collection_lists(self):
+        private_collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="test_user_id"
+        )
+        public_collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PUBLIC.name, owner="test_user_id"
+        )
+        collection_to_delete = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="test_user_id"
+        )
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.get("/dp/v1/collections/", headers=headers)
+
+        collection_ids = [collection["id"] for collection in json.loads(response.body)["collections"]]
+        self.assertIn(private_collection.id, collection_ids)
+        self.assertIn(public_collection.id, collection_ids)
+        self.assertIn(collection_to_delete.id, collection_ids)
+
```

Comment:
Remove, already tested in previous tests.

Reply:
see above comment